clippy_lints/src/minmax.rs

```diff
@@ -36,21 +37,43 @@ impl LintPass for MinMaxPass {
 impl<'a, 'tcx> LateLintPass<'a, 'tcx> for MinMaxPass {
     fn check_expr(&mut self, cx: &LateContext<'a, 'tcx>, expr: &'tcx Expr) {
         if let Some((outer_max, outer_c, oe)) = min_max(cx, expr) {
-            if let Some((inner_max, inner_c, _)) = min_max(cx, oe) {
+            if let Some((inner_max, inner_c, ie)) = min_max(cx, oe) {
                 if outer_max == inner_max {
                     return;
                 }
-                match (outer_max, outer_c.partial_cmp(&inner_c)) {
```

Comment:
Maybe we should replace the PartialCmp impl with an impl method that also takes a type (basically your code below). Because if we are using this method in other places, they probably have the same bug